stdlib/LinearAlgebra/src/tridiag.jl

```diff
@@ -693,6 +697,22 @@ end
     end
 end
 
+Base._reverse(A::Tridiagonal, dims) = reverse!(Matrix(A); dims)
+Base._reverse(A::Tridiagonal, dims::Colon) = Tridiagonal(reverse(A.du), reverse(A.d), reverse(A.dl))
+function Base._reverse!(A::Tridiagonal, dims::Colon)
+    n = length(A.du) # == length(A.dl), & always 1-based
```

Comment:
```suggestion
    n = length(A.du) # == length(A.dl), & always 1-based
    n == length(A.dl) || throw(DimensionMismatch("Diagonals have different lengths"))
    Base.require_one_based_indexing(A.dl, A.du)
```

From the docstring of `@inbounds`

> Only use `@inbounds` when it is certain from the information locally available that all accesses are in bounds.

The constructors of Tridiagonal are not locally available information, so they should be checked before relying on them in `@inbounds` below.

Reply:
I disagree — we know from locally available information that this is a `Tridiagonal` instance, and hence must obey the invariants of `Tridiagonal`.

(Otherwise, many of the other uses of `@inbounds` in tridiag.jl (and in many other files) are also wrong.)

Reply:
From the commentary on that PR it seems clear that @stevengj is right here.